config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -311,7 +324,19 @@ private static void writeValueData(DataOutput out, ConfigValue value) throws IOE
             out.writeUTF(((ConfigNumber) value).transformToString());
             break;
         case STRING:
-            out.writeUTF(((ConfigString) value).unwrapped());
+            String strVal = ((ConfigString) value).unwrapped();
+ 
+            List<String> values = new ArrayList<>();
+            if (strVal.getBytes().length >= MAX_BYTES_LENGTH) {
+                    values.addAll(split(strVal, MAX_BYTES_LENGTH/2)); // A char variable is typically 2-byte large.
```

Comment:
Shouldn't it be the maximum length instead of typical length? With `MAX_BYTES_LENGTH/2` it's still possible to construct the string which overflows the writeUTF limitations